applications/FluidDynamicsApplication/tests/cpp_tests/test_stokes_two_fluid_volume_error_source.cpp

```diff
@@ -0,0 +1,196 @@
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:         BSD License
+//                   Kratos default license: kratos/license.txt
+//
+//  Main authors:    Vincent Divine
+//
+//
+
+// System includes
+// External includes
+// Project includes
+#include "testing/testing.h"
+#include "containers/model.h"
+#include "includes/properties.h"
+#include "includes/model_part.h"
+#include "custom_elements/stokes_3D_twofluid.h"
+#include "custom_constitutive/newtonian_two_fluid_3d_law.h"
+#include "fluid_dynamics_application_variables.h"
+namespace Kratos {
+    namespace Testing {
+
+        typedef ModelPart::IndexType									 IndexType;
+        typedef ModelPart::NodeIterator					          NodeIteratorType;
+
+        void PrepareModelPartError(ModelPart& modelPart)
+        {
+            modelPart.SetBufferSize(3);
+            
+            // Variables addition
+            modelPart.AddNodalSolutionStepVariable(BODY_FORCE);
+            modelPart.AddNodalSolutionStepVariable(DENSITY);
+            modelPart.AddNodalSolutionStepVariable(DYNAMIC_VISCOSITY);
+            modelPart.AddNodalSolutionStepVariable(DYNAMIC_TAU);
+            modelPart.AddNodalSolutionStepVariable(PRESSURE);
+            modelPart.AddNodalSolutionStepVariable(VELOCITY);
+            modelPart.AddNodalSolutionStepVariable(MESH_VELOCITY);
+            modelPart.AddNodalSolutionStepVariable(DISTANCE);
+           
+
+            // Process info creation
+            double delta_time = 0.1;
+            modelPart.GetProcessInfo().SetValue(DYNAMIC_TAU, 0.001);
+            modelPart.GetProcessInfo().SetValue(DELTA_TIME, delta_time);
+            modelPart.GetProcessInfo().SetValue(VOLUME_ERROR, 10.0);
+            Vector bdf_coefs(3);
+            bdf_coefs[0] = 3.0 / (2.0*delta_time);
+            bdf_coefs[1] = -2.0 / delta_time;
+            bdf_coefs[2] = 0.5*delta_time;
+            modelPart.GetProcessInfo().SetValue(BDF_COEFFICIENTS, bdf_coefs);
+            // modelPart.GetProcessInfo().SetValue(VOLUME_ERROR,0);
+            // Set the element properties
+            Properties::Pointer pElemProp = modelPart.CreateNewProperties(0);
+            pElemProp->SetValue(DENSITY, 1000.0);
+            pElemProp->SetValue(DYNAMIC_VISCOSITY, 1.0e-05);
+            pElemProp->SetValue(DENSITY_AIR, 1.0);
+            NewtonianTwoFluid3DLaw::Pointer pConsLaw(new NewtonianTwoFluid3DLaw());
+            pElemProp->SetValue(CONSTITUTIVE_LAW, pConsLaw);
+
+            // Geometry creation
+            modelPart.CreateNewNode(1, 0.0, 0.0, 0.0);
+            modelPart.CreateNewNode(2, 1.0, 0.0, 0.0);
+            modelPart.CreateNewNode(3, 0.0, 1.0, 0.0);
+            modelPart.CreateNewNode(4, 0.0, 0.0, 1.0);
+            std::vector<ModelPart::IndexType> elemNodes{ 1, 2, 3, 4 };
+            modelPart.CreateNewElement("StokesTwoFluid3D4N", 1, elemNodes, pElemProp);
```

Comment:
Note that in here you're testing the 2-fluids Stokes element? (not the Navier-Stokes one). Besides, I wouldn't add a new file for these tests, I'd just add them to the already existent `test_two_fluid_navier_stokes_element.cpp`.

Reply:
You are right! Solved